lapce-core/src/syntax/mod.rs

```diff
@@ -841,11 +835,13 @@ impl Syntax {
             if start >= self.text.len() {
                 return None;
             }
-            let c = self.text.byte_at(start) as char;
-            if matching_pair_direction(c) == Some(true) {
-                let end = self.find_matching_pair(start)?;
-                if end >= offset {
-                    return Some((start, end));
+            if start < offset {
```

Comment:
Previously the the offset immediately before an opening bracket was considered enclosed (i.e.: `|(    )`). This was inconsistent with `WordCursor::find_enclosing_pair`, which is used for language unaware bracket search, so I've changed it to not consider that offset to be enclosed.